drivers/wifi/nrf700x/osal/fw_if/umac_if/src/event.c

```diff
@@ -527,13 +528,73 @@ wifi_nrf_fmac_data_event_process(struct wifi_nrf_fmac_dev_ctx *fmac_dev_ctx,
 
 	switch (event) {
 	case NRF_WIFI_CMD_RX_BUFF:
+#ifdef CONFIG_NRF700X_RX_DONE_WQ_ENABLED
+		struct nrf_wifi_rx_buff *config = wifi_nrf_osal_mem_zalloc(
+			fmac_dev_ctx->fpriv->opriv,
+			sizeof(struct nrf_wifi_rx_buff));
+		if (!config) {
+			wifi_nrf_osal_log_err(fmac_dev_ctx->fpriv->opriv,
+					      "%s: Failed to allocate memory (RX)\n",
+					      __func__);
+			status = WIFI_NRF_STATUS_FAIL;
+			break;
+		}
+		wifi_nrf_osal_mem_cpy(fmac_dev_ctx->fpriv->opriv,
+					config,
+					umac_head,
+					sizeof(struct nrf_wifi_rx_buff));
+		status = wifi_nrf_utils_q_enqueue(fmac_dev_ctx->fpriv->opriv,
+					 fmac_dev_ctx->rx_config.rx_tasklet_event_q,
+					 config);
+		if (status != WIFI_NRF_STATUS_SUCCESS) {
+			wifi_nrf_osal_log_err(fmac_dev_ctx->fpriv->opriv,
+					      "%s: Failed to enqueue RX buffer\n",
+					      __func__);
+			wifi_nrf_osal_mem_free(fmac_dev_ctx->fpriv->opriv,
+					       config);
+			break;
+		}
+		wifi_nrf_osal_tasklet_schedule(fmac_dev_ctx->fpriv->opriv,
+					       fmac_dev_ctx->rx_tasklet);
+#else
 		status = wifi_nrf_fmac_rx_event_process(fmac_dev_ctx,
 							umac_head);
+#endif /* CONFIG_NRF700X_RX_DONE_WQ_ENABLED */
 		break;
 #ifdef CONFIG_NRF700X_DATA_TX
 	case NRF_WIFI_CMD_TX_BUFF_DONE:
+#ifdef CONFIG_NRF700X_TX_DONE_WQ_ENABLED
+		struct nrf_wifi_tx_buff_done *config = wifi_nrf_osal_mem_zalloc(
+					fmac_dev_ctx->fpriv->opriv,
+					sizeof(struct nrf_wifi_tx_buff_done));
+		if (!config) {
+			wifi_nrf_osal_log_err(fmac_dev_ctx->fpriv->opriv,
+					      "%s: Failed to allocate memory (TX)\n",
+					      __func__);
+			status = WIFI_NRF_STATUS_FAIL;
+			break;
+		}
+		wifi_nrf_osal_mem_cpy(fmac_dev_ctx->fpriv->opriv,
+					config,
```

Comment:
Indentation

Reply:
Tried again, aligning to open brace vs. not exceeding 100 lines vs. using tabs as much as possible, its impossible to do this alignment anyother (AFAIK), so, closing this comment as checkpatch is passing.